packages/playground/src/utils/nodeSelector.ts

```diff
@@ -0,0 +1,193 @@
+import type { FarmFilterOptions, FarmInfo, FilterOptions, NodeInfo } from "@threefold/grid_client";
+import { NodeStatus } from "@threefold/gridproxy_client";
+import shuffle from "lodash/fp/shuffle.js";
+import type { Ref } from "vue";
+
+import { gqlClient, gridProxyClient } from "../clients";
+import type { useGrid } from "../stores";
+import type {
+  Locations,
+  NodeSelectorFilters,
+  NormalizeFarmFiltersOptions,
+  NormalizeNodeFiltersOptions,
+  SelectedLocation,
+} from "../types/nodeSelector";
+
+export async function getLocations(): Promise<Locations> {
+  const countries = await gqlClient.countries({ name: true, subregion: true });
+  const stats = await gridProxyClient.stats.get({ status: NodeStatus.Up });
+  const allowedCountriesList = Object.keys(stats.nodesDistribution);
+
+  const locations: Locations = {};
+  for (const country of countries) {
+    if (allowedCountriesList.includes(country.name)) {
+      locations[country.subregion] = locations[country.subregion] || [];
+      locations[country.subregion].push(country.name);
+    }
+  }
+  return locations;
+}
+
+export function normalizeFarmOptions(
+  gridStore: ReturnType<typeof useGrid>,
+  location: SelectedLocation | undefined,
+  page: Ref<number>,
+): NormalizeFarmFiltersOptions {
+  return { size: window.env.PAGE_SIZE, page: page.value, location: location || {}, twinId: gridStore.client.twinId };
+}
+
+export function normalizeFarmFilters(
+  filters: NodeSelectorFilters,
+  options: NormalizeFarmFiltersOptions,
+): FarmFilterOptions {
+  return {
+    size: options.size,
+    page: Math.max(1, options.page),
+    // availableFor: options.twinId,
```

Comment:
remove commented line.

Reply:
Commented for testing because it's very slow in mainnet